spec/defines/server/role_spec.rb

```diff
@@ -47,16 +47,18 @@
     it { is_expected.to contain_postgresql__server__role('test') }
     it 'has create role for "test" user with password as ****' do
       is_expected.to contain_postgresql_psql('CREATE ROLE test ENCRYPTED PASSWORD ****')
-        .with('command'     => 'Sensitive [value redacted]',
+        .with('command'     => sensitive(%(CREATE ROLE "test" ENCRYPTED PASSWORD 'new-pa$s' LOGIN NOCREATEROLE NOCREATEDB NOSUPERUSER  CONNECTION LIMIT -1)),
               'sensitive'   => 'true',
               'unless'      => "SELECT 1 FROM pg_roles WHERE rolname = 'test'",
               'port'        => '5432')
     end
     it 'has alter role for "test" user with password as ****' do
       is_expected.to contain_postgresql_psql('ALTER ROLE test ENCRYPTED PASSWORD ****')
-        .with('command'     => 'Sensitive [value redacted]',
+        .with('command'     => sensitive(%(ALTER ROLE "test" ENCRYPTED PASSWORD 'Sensitive [value redacted]')),
+              # FIXME: This is obviously wrong                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Any idea what's needed to fix this? Or is it actually a bug?

Reply:
yeah, the sensitive values definitely need unwrapping before interpolating into another string.

```puppet
$secret = Sensitive('SECRET')

exec { 'test':
  command => Sensitive("/bin/echo My secret is ${secret} > /tmp/test_secret"),
}
```

```shell
cat /tmp/test_secret
My secret is Sensitive [value redacted]
```

Reply:
Given the minimum supported version in the metadata is 6.24.0 it should be fine to just throw some `unwrap`s in there without even checking if the value is `Sensitive` first.

See https://github.com/puppetlabs/puppet/commit/41991b2dc65fd00f484523b915958bd82f0ff836

Reply:
Yeah, it is a bug.  I opened this PR in order to make it easier to diagnose the issues and fix it.  For that reason I do not change the tested value in this PR so that it can hopefully be merged quickly because it does not change the current (bad) behavior.  Yet, adding a comment for the obvious does not hurt.

The bug itself is tracked in #1404.